libs/state/src/lib/state-analytics-plugin/state-analytics-plugin.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable, Provider } from '@angular/core';
+import { GoogleAnalyticsService } from 'ngx-google-analytics';
+import { getActionTypeFromInstance, NgxsNextPluginFn, NgxsPlugin, NGXS_PLUGINS } from '@ngxs/store';
+
+/** Primitive types */
+const PRIMITIVE_TYPES = ['bigint', 'boolean', 'number', 'string'];
+
+/**
+ * Determines whether primitive is
+ * @param value
+ * @returns primitive
+ */
+function isPrimitive(value: unknown): value is bigint | boolean | number | string | null | undefined {
+  return value == null || PRIMITIVE_TYPES.includes(typeof value);
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StateAnalyticsPluginService implements NgxsPlugin {
+  /**
+   * Creates an instance of logger plugin.
+   * @param ga
+   */
+  constructor(private readonly ga: GoogleAnalyticsService) {}
+
+  /**
+   * Handles logger plugin
+   * @param state
+   * @param action
+   * @param next
+   * @returns
+   */
+  handle(state: unknown, action: unknown, next: NgxsNextPluginFn) {
+    console.log('In handle');
+    this.logAction(action);
+    return next(state, action);
+  }
+
+  /**
+   * Logs action trigerred to google analytics event
+   * @param action
+   * @returns action
+   */
+  private logAction(action: unknown): void {
+    const type = getActionTypeFromInstance(action);
+    console.log('type', type);
+
+    if (type) {
+      const payload = JSON.stringify(action, this.serialize);
+      console.log('payload', payload);
+      this.ga.event(type, payload);
+    }
+  }
+
+  /**
+   * Serializes action data
+   * @param this
+   * @param key
+   * @param value
+   * @returns serialize
+   */
+  private serialize(this: void, key: unknown, value: unknown): unknown {
```

Comment:
@bherr2 Should we update the serialization logic to include nested objects, etc.?

Reply:
First argument to ga.event (See <https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#ec>) has a 150 byte limit, second has a 500 byte limit, 3rd also has a 500 byte limit). If it goes over, it gets truncated.